BlazarUI/app/scripts/components/repo/RepoContainer.jsx

```diff
@@ -36,34 +36,32 @@ class RepoContainer extends Component {
   }
 
   componentDidMount() {
-    this.setup(this.props.params);
+    this.unsubscribeFromBuilds = BuildsStore.listen(this.onStatusChange);
+    BuildsActions.loadBuilds();
   }
 
   componentWillReceiveProps(nextprops) {
-    this.setState(initialState);
-    this.tearDown();
-    this.setup(nextprops.params);
+    this.setState({
+      filters: {
+        branch: [],
+        repo: []
+      },
+      error: null
+    });
   }
 
   componentWillUnmount() {
-    this.tearDown();
+    this.unsubscribeFromBuilds();
   }
 
   onStatusChange(state) {
     if (state.error) {
       state.loading = false;
     }
-    this.setState(state);
-  }
 
-  setup(params) {
-    this.unsubscribeFromRepo = RepoStore.listen(this.onStatusChange);
-    RepoActions.loadBranchesAndBuilds(params);
-  }
+    state.builds = state.builds.all;
```

Comment:
this component doesn't seem to need all the builds so you can do the filtering here instead of in the render method. I'm assuming that BranchFilter and BranchTable don't need access but it's hard to tell because of how {...this.state}} is passed into both.